pkg/agent/openflow/pipeline.go

```diff
@@ -1363,18 +1411,43 @@ func (f *featurePodConnectivity) l3FwdFlowToGateway() []binding.Flow {
 }
 
 // l3FwdFlowToRemoteViaTun generates the flow to match the packets destined for remote Pods via tunnel.
-func (f *featurePodConnectivity) l3FwdFlowToRemoteViaTun(localGatewayMAC net.HardwareAddr, peerSubnet net.IPNet, tunnelPeer net.IP) binding.Flow {
+func (f *featurePodConnectivity) l3FwdFlowToRemoteViaTun(localGatewayMAC net.HardwareAddr, peerSubnet net.IPNet, tunnelPeer net.IP) []binding.Flow {
 	ipProtocol := getIPProtocol(peerSubnet.IP)
-	return L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
-		Cookie(f.cookieAllocator.Request(f.category).Raw()).
-		MatchProtocol(ipProtocol).
-		MatchDstIPNet(peerSubnet).
-		Action().SetSrcMAC(localGatewayMAC).  // Rewrite src MAC to local gateway MAC.
-		Action().SetDstMAC(GlobalVirtualMAC). // Rewrite dst MAC to virtual MAC.
-		Action().SetTunnelDst(tunnelPeer).    // Flow based tunnel. Set tunnel destination.
-		Action().LoadRegMark(ToTunnelRegMark).
-		Action().GotoTable(L3DecTTLTable.GetID()).
-		Done()
+	buildFlow := func(matcher func(b binding.FlowBuilder) binding.FlowBuilder) binding.Flow {
+		builder := L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(f.cookieAllocator.Request(f.category).Raw()).
+			MatchProtocol(ipProtocol)
+		builder = matcher(builder)
+		return builder.
+			Action().SetSrcMAC(localGatewayMAC).  // Rewrite src MAC to local gateway MAC.
+			Action().SetDstMAC(GlobalVirtualMAC). // Rewrite dst MAC to virtual MAC.
+			Action().SetTunnelDst(tunnelPeer).    // Flow based tunnel. Set tunnel destination.
+			Action().LoadRegMark(ToTunnelRegMark).
+			Action().GotoTable(L3DecTTLTable.GetID()).
+			Done()
+	}
```

Comment:
I'm okay with that. I prefer that every flow is written in its own code block if possible. Since there are only a few differences between these flows, it should be a good way as yours.